---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
version
#sessionInfo()
#View(SampleTable)
```


```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("DESeq2")
```

```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("apeglm")
```


```{r}

directory <- "C:/Users/holly/OneDrive/Documents/GitHub/Genome_Analysis/6_RNA_mapping/output"

#sampleFiles <- grep("count",list.files(directory),value=TRUE)

sampleFiles<-c('rna_bh_ERR1797972_counts.txt', 'rna_bh_ERR1797973_counts.txt', 'rna_bh_ERR1797974_counts.txt', 'rna_serum_ERR1797969_counts.txt', 'rna_serum_ERR1797970_counts.txt', 'rna_serum_ERR1797971_counts.txt')
sampleCondition<- c('control', 'control', 'control', 't1', 't2', 't3')
#sampleCondition <- sub("(.*count.txt).*","\\1",sampleFiles)
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)
```

```{r}
library("DESeq2")
ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                       directory = directory,
                                       design= ~ condition)
ddsHTSeq
```

Filtered out genes with reads less than 10
```{r}
dds<-DESeq(ddsHTSeq)
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```


resLFC data were transformed to show log-fold changes which are often easier to visualize

```{r}

#res<-results(dds)
res1 <- results(dds, name="condition_t1_vs_control", alpha=0.05)


resLFC1 <- lfcShrink(dds, coef="condition_t1_vs_control", type="apeglm")
#this will give a results table of log fold changes (LFC) of T over C. Here a positive LFC means T has higher counts than C. A negative LFC means T has lower counts than C.

res2 <- results(dds, name="condition_t2_vs_control", alpha=0.05)


resLFC2 <- lfcShrink(dds, coef="condition_t2_vs_control", type="apeglm")

res3 <- results(dds, name="condition_t3_vs_control", alpha=0.05)


resLFC3 <- lfcShrink(dds, coef="condition_t3_vs_control", type="apeglm")

```

Results are ordered by their Pvalue
```{r}
#res1
res1Ordered <- res1[order(res1$padj),]
res2Ordered <- res2[order(res2$padj),]
res3Ordered <- res3[order(res3$padj),]

resLFC1Ordered <- resLFC1[order(resLFC1$padj),]
resLFC2Ordered <- resLFC2[order(resLFC2$padj),]
resLFC3Ordered <- resLFC3[order(resLFC3$padj),]


```

```{r}
resLFC1
```


Top results of differentially expressed RNA from human serum
```{r}
n = 100
topResults <- rbind( res1Ordered[ abs(res1Ordered[,'log2FoldChange']) > 2, ][1:n,], 
                     res1Ordered[ abs(res1Ordered[,'log2FoldChange']) < 0.5, ][n:1,] )
topResults[c(1:5,(2*n-4):(2*n)), c('baseMean','log2FoldChange','padj', 'pvalue')]

topResults2 <- rbind( res2Ordered[ abs(res2Ordered[,'log2FoldChange']) > 2, ][1:n,], 
                     res2Ordered[ abs(res2Ordered[,'log2FoldChange']) < 0.5, ][n:1,] )
topResults2[c(1:5,(2*n-4):(2*n)), c('baseMean','log2FoldChange','padj', 'pvalue')]

topResults3 <- rbind( res3Ordered[ abs(res3Ordered[,'log2FoldChange']) > 2, ][1:n,], 
                     res3Ordered[ abs(res3Ordered[,'log2FoldChange']) < 0.5, ][n:1,] )
topResults3[c(1:5,(2*n-4):(2*n)), c('baseMean','log2FoldChange','padj', 'pvalue')]

```



MA plots of log-fold changes
```{r}
plotMA(res1Ordered, ylim = c(-7, 7))
plotMA(topResults, ylim = c(-7, 7))

plotMA(resLFC1, ylim = c(-6, 6))
plotMA(resLFC2, ylim = c(-6, 6))
plotMA(resLFC3, ylim = c(-6, 6))
```


Genes with most significan expression ordered by adjusted p-value for each sequencing run of RNA from human serum.
```{r}
topResultsLFC1 <- rbind( resLFC1Ordered[ abs(resLFC1Ordered[,'log2FoldChange']) > 2, ][1:n,], 
                     resLFC1Ordered[ abs(resLFC1Ordered[,'log2FoldChange']) < 0.5, ][n:1,] )
topResultsLFC1[c(1:5,(2*n-4):(2*n)), c('baseMean','log2FoldChange','padj', 'pvalue')]

topResultsLFC2 <- rbind( resLFC2Ordered[ abs(resLFC2Ordered[,'log2FoldChange']) > 2, ][1:n,], 
                     resLFC2Ordered[ abs(resLFC2Ordered[,'log2FoldChange']) < 0.5, ][n:1,] )
topResultsLFC2[c(1:5,(2*n-4):(2*n)), c('baseMean','log2FoldChange','padj', 'pvalue')]

topResultsLFC3 <- rbind( resLFC3Ordered[ abs(resLFC3Ordered[,'log2FoldChange']) > 2, ][1:n,], 
                     resLFC3Ordered[ abs(resLFC3Ordered[,'log2FoldChange']) < 0.5, ][n:1,] )
topResultsLFC3[c(1:5,(2*n-4):(2*n)), c('baseMean','log2FoldChange','padj', 'pvalue')]
```



Observing and explaining the Variance
```{r}

pca <- rlogTransformation(dds, blind=TRUE)
plotPCA(pca)

vsd<-vst(dds)
plotPCA(vsd)
```





```{r}
# Genes normalized counts heatmap
library("RColorBrewer")
 colmn <- brewer.pal(9,"Blues")
 Counts <- counts(dds, normalized=TRUE)
 heatmap(as.matrix(Counts[ row.names(topResults), ]), Rowv = NA, col = colmn, mar = c(8,2))

```

